electron/legendary_utils/library.ts

```diff
@@ -56,85 +56,161 @@ export async function getLegendaryConfig(file: string): Promise<unknown> {
+            const {
+              description,
+              keyImages,
+              title,
+              developer,
+              releaseInfo
+            } = metadata
+
+            const gameBox = keyImages.filter(
+              ({ type }: KeyImage) => type === 'Screenshot'
+            )[0]
+            const gameBoxTall = keyImages.filter(
+              ({ type }: KeyImage) => type === 'Screenshot'
+            )[0]
+            const logo = keyImages.filter(
+              ({ type }: KeyImage) => type === 'Thumbnail'
+            )[0]
+
+            let compatibleApps
+            releaseInfo.forEach(
+              (rI: { appId : string, compatibleApps : string[] } ) => {
+                if (rI.appId == app_name) {
+                  compatibleApps = rI.compatibleApps
                 }
               }
             )
-          }
 
-          const cloudSaveEnabled = Boolean(CloudSaveFolder)
-          const saveFolder = cloudSaveEnabled ? CloudSaveFolder.value : ''
-          const installFolder = FolderName ? FolderName.value : ''
-          const gameBox = keyImages.filter(
-            ({ type }: KeyImage) => type === 'DieselGameBox'
-          )[0]
-          const gameBoxTall = keyImages.filter(
-            ({ type }: KeyImage) => type === 'DieselGameBoxTall'
-          )[0]
-          const logo = keyImages.filter(
-            ({ type }: KeyImage) => type === 'DieselGameBoxLogo'
-          )[0]
-
-          const art_cover = gameBox ? gameBox.url : null
-          const art_logo = logo ? logo.url : null
-          const art_square = gameBoxTall ? gameBoxTall.url : fallBackImage
-
-          const installedGames: Game[] = Object.values(files.installed)
-
-          const isInstalled = Boolean(
-            installedGames.filter((game) => game.app_name === app_name).length
-          )
-          const info = isInstalled
-            ? installedGames.filter((game) => game.app_name === app_name)[0]
-            : {}
-
-          const dlc = () => dlcs.some((dlc) => dlc === app_name)
-          const {
-            executable = null,
-            version = null,
-            install_size = null,
-            install_path = null,
-            is_dlc = dlc()
-          } = info as InstalledInfo
-
-          const convertedSize =
-            install_size &&
-            `${byteSize(install_size).value}${byteSize(install_size).unit}`
-
-          return {
-            app_name,
-            art_cover: art_cover || art_square,
-            art_logo,
-            art_square: art_square || art_cover,
-            cloudSaveEnabled,
-            description,
-            developer,
-            executable,
-            folderName: installFolder,
-            info,
-            install_path,
-            install_size: convertedSize,
-            isInstalled,
-            is_dlc,
-            namespace,
-            saveFolder,
-            title,
-            version
+            const art_cover = gameBox ? gameBox.url : null
```

Comment:
I noticed that this part of the code is repeated almost entirely, we could maybe have a helper function to that, or check the namespace inside it. You could have a variable called: `isUEAsset`, for instance, and use it where you need to check different things.